src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEMethodSymbol.cs

```diff
@@ -1221,6 +1215,38 @@ public override ImmutableArray<MethodSymbol> ExplicitInterfaceImplementations
             }
         }
 
+        private ImmutableArray<MethodSymbol> ExplicitlyOverriddenMethods()
+        {
+            var moduleSymbol = _containingType.ContainingPEModule;
+
+            // Context: we need the containing type of this method as context so that we can substitute appropriately into
+            // any generic interfaces that we might be explicitly implementing.  There is no reason to pass in the method
+            // context, however, because any method type parameters will belong to the implemented (i.e. interface) method,
+            // which we do not yet know.
+            return new MetadataDecoder(moduleSymbol, _containingType).GetExplicitlyOverriddenMethods(_containingType.Handle, _handle, this.ContainingType);
+        }
+
+        internal MethodSymbol ExplicitlyOverriddenClassMethod
+        {
+            get
+            {
+                if (IsExplicitClassOverride)
+                {
+                    foreach (MethodSymbol method in ExplicitlyOverriddenMethods())
+                    {
+                        if (method.ContainingType.IsClassType())
+                        {
+                            return method;
```

Comment:
>return method; [](start = 28, length = 14)

It doesn't feel appropriate to assume that there is only one explicit override like that or depend on order of MethodImps in metadata.

Reply:
I will make this return `null` when there is more than one in the next PR #44025.  The C# compiler will only produce one, at least today. If in the future that changes we may need to change the shape of this method and the API `OverriddenMethod`.

---
In reply to: [421733592](https://github.com/dotnet/roslyn/pull/43797#discussion_r421733592) [](ancestors = 421733592)